aws/resource_aws_dms_endpoint.go

```diff
@@ -117,6 +120,14 @@ func resourceAwsDmsEndpoint() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 			},
+			"bucket_name": {
```

Comment:
To better match the AWS API and SDK, we should nest these attributes under a new configuration block:

```go
"s3_settings": {
  Type:     schema.TypeList,
  Optional: true,
  MaxItems: 1,
  Elem: &schema.Resource{
    Schema: map[string]*schema.Schema{
      "bucket_name": {
        Type:     schema.TypeString,
        Required: true,
      },
      "bucket_folder": {
        Type:     schema.TypeString,
        Optional: true,
      },
    },
  },
```

This will allow us to also implement other attributes of the S3 configuration better such as `compression_type`, `csv_delimiter`, and `csv_row_delimiter` without polluting the top attribute namespace.

Reply:
Good idea
Will implement it also

Reply:
@bflad 
I have checked it. If we will implement it like this, then what should we do with extra_connection_attributes attribute? Because even if it is not like that in AWS API, but it is like this in AWS UI. Only bucket_name and folder is separated in UI.